paimon-core/src/test/java/org/apache/paimon/operation/KeyValueFileStoreWriteTest.java

```diff
@@ -102,33 +125,166 @@ public void testGentleLookupCompactStrategy() throws Exception {
         }
     }
 
-    private KeyValueFileStoreWrite createWriteWithOptions(Map<String, String> options)
-            throws Exception {
-        SchemaManager schemaManager =
-                new SchemaManager(LocalFileIO.create(), new Path(tempDir.toUri()));
+    @Test
+    public void testMultiWriteModeEnabled() throws Exception {
+        Map<String, String> options1 = new HashMap<>();
+        options1.put(CoreOptions.WRITE_MODE.key(), "multi_write");
+        options1.put(CoreOptions.NUM_SORTED_RUNS_COMPACTION_TRIGGER.key(), "2");
+        String user1 = "user1";
+        String user2 = "user2";
 
-        TableSchema schema =
-                schemaManager.createTable(
-                        new Schema(
-                                TestKeyValueGenerator.DEFAULT_ROW_TYPE.getFields(),
-                                TestKeyValueGenerator.DEFAULT_PART_TYPE.getFieldNames(),
-                                TestKeyValueGenerator.getPrimaryKeys(
-                                        TestKeyValueGenerator.GeneratorMode.MULTI_PARTITIONED),
-                                options,
-                                null));
-        TestFileStore store =
-                new TestFileStore.Builder(
-                                "avro",
-                                tempDir.toString(),
-                                NUM_BUCKETS,
-                                TestKeyValueGenerator.DEFAULT_PART_TYPE,
-                                TestKeyValueGenerator.KEY_TYPE,
-                                TestKeyValueGenerator.DEFAULT_ROW_TYPE,
-                                TestKeyValueGenerator.TestKeyValueFieldsExtractor.EXTRACTOR,
-                                DeduplicateMergeFunction.factory(),
-                                schema)
-                        .build();
+        Map<String, String> options2 = new HashMap<>();
+        options2.put(CoreOptions.WRITE_ONLY.key(), "true");
+
+        TestFileStore store1 = createStoreWithOptions(options1);
+        TestFileStore store2 = createStoreWithOptions(options2);
+
+        KeyValueFileStoreWrite write1 = (KeyValueFileStoreWrite) store1.newWrite(user1);
+        KeyValueFileStoreWrite write2 = (KeyValueFileStoreWrite) store2.newWrite(user2);
+
```

Comment:
I tried adding a writer-3 in this test, the result is not as expected, the file written by the writer-1 is not be compacted.

Reply:
writer-1 is the write and compact job, writer-3 should be write only. The file added by writer-3 should be compacted by writer-1.

Reply:
Sorry, my expression is wrong.  the file written by the writer-2 is not be compacted. write-1 and write-3 will be compacted.

Reply:
@LinMingQiang There is a debug in the previous implementable. Pls check the latest version. Also I've added the writer-3 in the test.